corehq/apps/user_importer/importer.py

```diff
@@ -527,60 +545,108 @@ def create_or_update_users_and_groups(upload_domain, user_specs, upload_user, gr
 
                     # Do not update location info if the column is not included at all
                     location_ids = find_location_id(location_codes, domain_info.location_cache)
+                    users_current_primary_location_id = user.location_id
                     locations_updated, primary_loc_removed = check_modified_user_loc(location_ids,
                                                                                      user.location_id,
                                                                                      user.assigned_location_ids)
                     if primary_loc_removed:
                         user.unset_location(commit=False)
                     if locations_updated:
                         user.reset_locations(location_ids, commit=False)
-
+                        user_change_logger.add_changes({'assigned_location_ids': location_ids})
+                        if location_ids:
+                            location_names = list(SQLLocation.active_objects.filter(
```

Comment:
can you not get this from `domain_info.location_cache` instead of doing another lookup?

Reply:
Also wondering if this couldn't be moved to the logger to make the logging less 'invasive' on the import code:
```
user_change_logger.add_location_change_info(location_ids, domain_info.location_cache)
```

Reply:
thank you, totally missed about the caching importer has handy. Will incorporate that.